### DISCUSSION TITLE
Unable to add database

### DISCUSSION THREAD

**Discussion by JP021993** [pawelsalawa/sqlitestudio#4279] [Q&A]
### Details

Downloaded **Sqlite Studio 3.3.3** - Windows 10 (64 bit)

I'm unable to add any type of database - I tried to test the connection and every time it shows a red exclamation point. Status at bottom of Sqlite window shows "could not add database," but no reason. I'm new to this, so I have no idea what I need to include here to help you.

One thing I noticed is that Sqlite studio shows as an application under the downloaded zip file, but doesn't show as a stand alone application anywhere else on my computer. I extracted all, so I could open Sqlite studio, but that's all I'm able to do with it.

### Steps to reproduce
Restarted computer
Closed - restarted app
Attempted to play with some permissions, but I've never had any issues with permissions before and I'm the only user on this computer.

Here's info from my system settings (I removed Device ID): 

Device name	LAPTOP-Q8K8AA6E
Processor	AMD A12-9700P RADEON R7, 10 COMPUTE CORES 4C+6G   2.50 GHz
Installed RAM	6.00 GB (5.59 GB usable)
Product ID	00325-80764-62878-AAOEM
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

Edition	Windows 10 Home
Version	21H1
Installed on	‎9/‎10/‎2020
OS build	19043.1288
Experience	Windows Feature Experience Pack 120.2212.3920.0

**Comment by pawelsalawa** [comment]
1. It doesn't show as "standalone" application, because version 3.3.x is available only as a portable package (zip package, that you extract and run from there, no installation). Version 3.4.x will bring back the installer (as an option).
2. Are you trying to add existing database file, or to create new one?
3. If it's an existing file, then it maybe is not a valid SQLite database? Or it was created with newer SQLite version than the one that is used in SQLiteStudio 3.3.3? Or maybe it's encrypted with the password? (in which case you need to choose different database type first, to indicate encryption algorithm and then type in the password)
4. If you're trying to add new database file (not existing yet), then maybe you don't have permission to create files in the destination folder? Generally creating new database should work in most cases.

**Comment by JP021993** [comment]
Okay. Do you know where I can find version 3.4? I didn’t see it on what I think is the main website. Sqlitestudio.pl — another website I found was SQLite.org but that looked overly complex to download.

I’m adding an existing database from a LinkedIn Learning class. The version he used was 3.2.1 - does that automatically disqualify the files and/or databases?

No password is needed. It’s a free class online and the text version of the information (not the databases) opens easily in notepad++ (a suped up text editor)

thank you for the help!

**Comment by pawelsalawa** [comment]
3.4 is in development. Not released yet.

sqlite.org is page of the SQLite database engine - set of algorithms that can understand, read & write SQLite databases.
sqlitestudio.pl is application, editor for SQLite databases. It uses SQLite engine underneath.

Is your database file perhaps compressed with zip or rar?

**Reply by JP021993** [reply]
It’s compressed with a zip, but I extracted all and had the same issue.

**Reply by pawelsalawa** [reply]
Is it possible that you share the database file with me? Either be attaching it to the post here (if you don't mind having it publicly available), or by sending to me to support@sqlitestudio.pl?

If not, I'm not sure how else I can help.

**Reply by JP021993** [reply]
Yes, I emailed the zipped version to you.

**Reply by pawelsalawa** [reply]
![recording](https://user-images.githubusercontent.com/481846/139598397-53b79dd7-300e-49b2-8e37-ef58249e30ff.gif)

**Reply by pawelsalawa** [reply]
All 3 databases seem to open just fine. Doesn't it work the same way for you?

**Comment by JP021993** [comment]
Just realized you probably can't see that. For some reason the app isn't
visible in the video.

**Comment by JP021993** [comment]
I tried to send you a video, but none of the options were working. Here's
the error I receive on all attempts to upload the databases though:

[16:13:34] Could not add database
C:/Users/Josiah/Documents/Ex_Files_SQL_EssT/Ex_Files_SQL_EssT/Exercise
Files/SQL/album-sqlite3.sql:

**Reply by pawelsalawa** [reply]
SQL files are not databases. They are SQL scripts. You can open them even with Notepad to see SQL queries inside.

Databases are in `db` folder and have extension `db`.

**Comment by JP021993** [comment]
Yeah, see I told you I was new. Any idea what may be happening?

**Reply by pawelsalawa** [reply]
It's happening, because you're trying to open SQL files, which are not databases.

When opening a database, browse for *.db files.

**Reply by pawelsalawa** [reply]
Watch the video above again.

**Comment by JP021993** [comment]
Wow. I apologize for that.
